pkg/util/labels/labels_test.go

```diff
@@ -148,11 +149,20 @@ func TestCloneSelectorAndAddLabel(t *testing.T) {
 				"foo4": "12",
 			},
 		},
+		{
+			labels:           nil,
+			labelKey:         "foo4",
+			labelValue:       "89",
+			matchExpressions: []metav1.LabelSelectorRequirement{{Key: "foo1", Operator: "in", Values: []string{"bar1"}}},
```

Comment:
why just this one case and not all the interesting cases?

Reply:
I am with @thockin here, except that you don't need to be exhaustive. If you can cover some interesting cases to make sure that `matchExpressions` are preserved untouched irrespective of the `labelKey`s and `labelValue`s provided it is sufficient. Particularly, check whether `matchExpressions` are untouched even if you are adding `labelKey`s and `labelValue`s that are already in `matchExpressions`.

Reply:
@thockin @madhusudancs sure thing! I will get those added.

Reply:
code freeze is coming.  Reassign me if you need me.